packages/styled-base/src/index.js

```diff
@@ -72,99 +74,87 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
-              }
-            }
-
-            newProps.className = className
-
-            newProps.ref = ref || props.innerRef
-            if (process.env.NODE_ENV !== 'production' && props.innerRef) {
-              console.error(
-                '`innerRef` is deprecated and will be removed in a future major version of Emotion, please use the `ref` prop instead' +
-                  (identifierName === undefined
-                    ? ''
-                    : ` in the usage of \`${identifierName}\``)
-              )
-            }
-
-            const ele = React.createElement(finalTag, newProps)
-            if (!isBrowser && rules !== undefined) {
-              let serializedNames = serialized.name
-              let next = serialized.next
-              while (next !== undefined) {
-                serializedNames += ' ' + next.name
-                next = next.next
-              }
-              return (
-                <React.Fragment>
-                  <style
-                    {...{
-                      [`data-emotion-${context.key}`]: serializedNames,
-                      dangerouslySetInnerHTML: { __html: rules },
-                      nonce: context.sheet.nonce
-                    }}
-                  />
-                  {ele}
-                </React.Fragment>
-              )
-            }
-            return ele
-          }}
-        </ThemeContext.Consumer>
+      const finalTag = (shouldUseAs && props.as) || baseTag
+      let className = ''
+      let classInterpolations = []
+      let mergedProps = props
+      if (props.theme == null) {
+        mergedProps = {}
+        for (let key in props) {
+          mergedProps[key] = props[key]
+        }
+        mergedProps.theme = useContext(ThemeContext)
+      }
+      if (typeof props.className === 'string') {
+        className += getRegisteredStyles(
+          context.registered,
+          classInterpolations,
+          props.className
+        )
+      }
+      const serialized = serializeStyles(
+        styles.concat(classInterpolations),
+        context.registered,
+        mergedProps
+      )
+      const rules = insertStyles(
+        context,
+        serialized,
+        typeof finalTag === 'string'
       )
+      className += `${context.key}-${serialized.name}`
+      if (targetClassName !== undefined) {
+        className += ` ${targetClassName}`
+      }
+
+      const finalShouldForwardProp =
+        shouldUseAs && shouldForwardProp === undefined
+          ? getDefaultShouldForwardProp(finalTag)
+          : defaultShouldForwardProp
+
+      let newProps = {}
+
+      for (let key in props) {
+        if (shouldUseAs && key === 'as') continue
+        if (finalShouldForwardProp(key)) {
+          newProps[key] = props[key]
+        }
+      }
+
+      newProps.className = className
+
+      newProps.ref = ref || props.innerRef
+
+      if (process.env.NODE_ENV !== 'production' && props.innerRef) {
```

Comment:
as this can't land without bumping major version I think we can remove support of innerRef here altogether